File: backend/components/metiers/administration_publique/guides/helpers/accessibility/helpers_accessibility.test.py
# helpers_accessibility.test.py – Tests unitaires et edge cases Python
from .helpers_accessibility import check_accessibility, audit_accessibility

def test_check_accessibility():
    assert check_accessibility({'contrast': 8, 'keyboard': True}) is True
    assert check_accessibility({'contrast': 2, 'keyboard': False}) is False

def test_audit_accessibility():
    result = audit_accessibility({'contrast': 8, 'keyboard': True})
    assert isinstance(result, dict)
    assert 'score' in result
    assert 'compliant' in result

File: backend/components/metiers/administration_publique/guides/helpers/accessibility/helpers_accessibility.test.js
// helpers_accessibility.test.js – Tests unitaires et edge cases JS
const { checkAccessibility, auditAccessibility } = require('./helpers_accessibility');

test('checkAccessibility returns true for compliant', () => {
  expect(checkAccessibility({ contrast: 8, keyboard: true })).toBe(true);
  expect(checkAccessibility({ contrast: 2, keyboard: false })).toBe(false);
});

test('auditAccessibility returns audit object', () => {
  const result = auditAccessibility({ contrast: 8, keyboard: true });
  expect(typeof result).toBe('object');
  expect(result).toHaveProperty('score');
  expect(result).toHaveProperty('compliant');
});
